functable.h

```diff
@@ -13,6 +13,9 @@ struct functable_s {
     Pos      (* quick_insert_string)(deflate_state *const s, const uint32_t str);
     uint32_t (* adler32)            (uint32_t adler, const unsigned char *buf, size_t len);
     uint32_t (* crc32)              (uint32_t crc, const unsigned char *buf, uint64_t len);
+#ifdef S390_VX_CRC32
```

Comment:
Why expose the original function? For testing? I don't know if that is something we want to do.

Reply:
It's needed for non-aligned portions of the input buffer. Since it can be either `crc32_big` or `crc32_generic`, I've decided not to duplicate the selection logic in the vectorized function, but rather simply call what `crc32_stub` has chosen.

Reply:
Indirect call is a lot slower than calling the correct function directly... if system can be both big and little endian, there is already macros to test which one should be called.

Reply:
It's not about the endianness in this case, but rather about whether `sizeof(void *) == sizeof(ptrdiff_t)` holds. With gcc and clang it most likely always does, but I wanted to be conservative here. I can just call `crc32_generic` - it's used for less than 64 bytes anyway.

Reply:
@iii-i If that test fails, it shouldn't even be using vector-optimized versions. That is to make sure there isn't any unexpected behavior due to compiler behaving non-standard.

Reply:
If the two types have different bit sizes, it can cause overflow or wraparound.